### ISSUE TITLE
Big Sur icon not applying in Dock after v1.55.0 update

### ISSUE THREAD

**Issue by mattbanks** [microsoft/vscode#120262]
Issue Type: <b>Bug</b>

I'm seeing the updated icon in the Applications folder, but in the Dock I still see the old icon. Tried quitting/restarting the app multiple times and also removing the app from the Dock and re-adding.

macOS 11.2.3 on 16" MacBook Pro

VS Code version: Code 1.55.0 (c185983a683d14c396952dd432459097bc7f757f, 2021-03-30T16:01:05.981Z)
OS version: Darwin x64 20.3.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 2, 2|
|Memory (System)|16.00GB (0.50GB free)|
|Process Argv|--crash-reporter-id 437e32fe-7fed-4693-873a-02b95a5454ed|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (36)</summary>

Extension|Author (truncated)|Version
---|---|---
rainbow-brackets|2gu|0.0.6
better-comments|aar|2.1.0
vscode-apollo|apo|1.18.2
better-toml|bun|0.3.2
npm-intellisense|chr|1.3.1
path-intellisense|chr|2.3.0
bracket-pair-colorizer-2|Coe|0.2.0
vscode-eslint|dba|2.1.19
es7-react-js-snippets|dsz|3.1.1
gitlens|eam|11.3.0
EditorConfig|Edi|0.16.4
prettier-vscode|esb|6.3.1
auto-close-tag|for|0.5.10
auto-rename-tag|for|0.1.6
go|gol|0.23.3
vscode-graphql|Gra|0.3.15
phpcs|ika|1.0.5
vscode-styled-components|jpo|1.5.0
dotenv|mik|1.0.1
vscode-scss|mrm|0.9.0
vscode-docker|ms-|1.11.0
atom-keybindings|ms-|3.0.9
vscode-typescript-tslint-plugin|ms-|1.3.3
debugger-for-chrome|msj|4.12.12
color-highlight|nau|2.3.0
vetur|oct|0.33.1
vscode-jest|Ort|3.2.0
material-icon-theme|PKi|4.6.0
linter-xo|sam|2.3.3
autoimport|ste|1.5.3
code-spell-checker|str|1.10.2
vscode-stylelint|sty|0.86.0
wordpress-snippet|tun|1.1.5
JavaScriptSnippets|xab|1.8.0
material-theme|zhu|3.9.15
html-css-class-completion|Zig|1.20.0

</details>

**Comment by lramos15** [comment]
I believe I saw this too when I first experienced it. It took a little bit of time for the cache on my machine to update. Have you tried restarting your machine?

**Comment by mattbanks** [comment]
Yeah, I restarted and still seeing the old icon. Confirmed on my 15" MacBook Pro as well (work and personal machines).

**Comment by misolori** [comment]
Yea, this has to do with the icon cache. See https://github.com/microsoft/vscode/issues/101014#issuecomment-808368475 for clearing the cache:

>## Clearing Icon Cache for Stable
>
>Depending on your platform, you may still see the earlier logo due to operating system caching of the application icon. Here are a few methods for the Mac that you can attempt to use to clear the cache:
>
>1. Restart your dock via `killall Dock` in the terminal
>
>2. Run these scripts in your terminal (which updates the timestamp of the icons):
>```
>sudo touch /Applications/Visual\ Studio\ Code.app;
>sudo touch /Applications/Visual\ Studio\ Code.app/Contents/Info.plist;
>sudo touch /Applications/Visual\ Studio\ Code.app/Contents/Resources/Code.icns;
>killall Dock Finder
>```
>_Note: For Insiders and Explorations, simply append `\ -\ Insiders` or `\ -\ Exploration` to each version name reference._
>
>3. Force the IconService cache to be rebuilt [using these steps](https://mjtsai.com/blog/2019/02/01/clearing-the-icon-services-cache-in-mojave/)

**Comment by mattbanks** [comment]
That did it (step 2 was the one that did the trick for me)!
